W-13539925 TLS/Redis shared storage docs (#331)
mulesoft/docs-gateway: gateway/1.5/modules/ROOT/pages/flex-local-configuration-reference-guide.adoc

spec:
 sharedStorage:
   redis:
     address: <string> // REQUIRED
     username: <string> // OPTIONAL
     password: <string> // OPTIONAL
     db: <string> // OPTIONAL
     tls: // OPTIONAL

>        trustedCA: <string> // REQUIRED

Reviewer: ```suggestion
       trustedCA: <string> // OPTIONAL
```

>        certificate: // REQUIRED

Reviewer: ```suggestion
       certificate: // OPTIONAL
```

|`sharedStorage.redis.address`
|Required
|N/A
|The Redis instance address in `host:port` format

|`sharedStorage.redis.username`
|Optional
|Empty
|The Redis user name

|`sharedStorage.redis.password`
|Optional
|Empty
|The Redis user password

|`sharedStorage.redis.db`
|Optional 
|0
|The Redis database number to use

|`sharedStorage.redis.tls`
|Optional
|Empty
|The Redis TLS configuration

| `sharedStorage.redis.tls.trustedCA`
| Required
| N/A

> | The client certificate used in TLS

Reviewer: This is the same as "truststore" in the connected mode
```suggestion
| Root certificate authority to use when verifying server certificates
```

| `sharedStorage.redis.tls.certificate`
| Required
| N/A

> | A TLS certificate key pair

Reviewer: ```suggestion
| Client certificate to present to the server
```

| `sharedStorage.redis.tls.keyPassphrase`
| Optional
| `none`

> | The certificate key passphrase. If specified, use the private key format that includes a DEK-Info header. If unspecified, the private key is not encrypted.

Reviewer: The customer could still provide an encrypted private key, but it will fail if the passphrase is missing

```suggestion
| The certificate key passphrase. If specified, use the private key format that includes a DEK-Info header. If unspecified, the private key must not be encrypted.
```

| `sharedStorage.redis.tls.certificate.key`
| Required
| N/A
| The private key part of the certificate

| `sharedStorage.redis.tls.certificate.crt`
| Required
| N/A
| The public key part of the certificate

|`sharedStorage.redis.tls.skipValidation`
| Optional
| `false`

> |

Reviewer: ```suggestion
| If skipValidation is true, any certificate presented by the server is accepted
```

| `sharedStorage.redis.tls.minversion`
| Optional
| `1.2`
| The minimum TLS version allowed

| `sharedStorage.redis.tls.maxversion`
| Optional
| `1.3`
| The maximum TLS version allowed

| `sharedStorage.redis.tls.alpn`
| Optional

> | `h2` and `http/1.1`

Reviewer: The default is "empty list" in this case

```suggestion
| 
```

     password: ecomm-pwd-123
     DB: 7
----
====

.*Redis with TLS Shared Storage Example*
[%collapsible]
====
The Redis storage service is an implementation of a REST API object store that functions as a bridge between the object store interface and a Redis backend. Optionally, you can use Redis with TLS to protect data and ensure secure communication.

> 

Reviewer: This is an internal implementation detail of Flex, I don't think customers should be aware of it
